### PR TITLE
FileSystem RW

### PR DESCRIPTION
**Pull request by ghost**
Completely untested, but writes FSTAB to that of a R/W filesystem. This **Should** work on watchOS 3.x, although it can be modified for 4.x. If you'd like, I can remove the compatibility stuff from readme if you want.

May or may not work, and will **break** support on 4.x.

### CODE REVIEW

**Comment by nullpixel** [comment]
there is so much stuff wrong with this it's actually unreal

File: OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m [hunk 1]
```diff
@@ -113,20 +115,36 @@ - (IBAction)runTheTingButton {
...
+    /dev/disk0s1s2 /private/var hfs,nosuid,nodev rw 0 2
+    
+    @What you need to mount FS as RW.
+
+    */
+NSError *error;
+NSString *stringToWrite = @"/dev/disk0s1s1 / hfs rw 0 1\n/dev/disk0s1s2 /private/var hfs,nosuid,nodev rw 0 2\n";
+NSString *filePath = [[NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject] stringByAppendingPathComponent:@"/User/fstab"];
+[stringToWrite writeToFile:filePath atomically:YES encoding:NSUTF8StringEncoding error:&error];
+system("chmod 777 /private/etc/fstab") // yes, i can use chmod(), but I'm more conformatatble using system(), as chmod() might be ineffective outside of the sandbox. (It may not be actual chmod())
```
Comment (stek29) [inline OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:140]: wat. there's no system on watchOS, and there's no way system(chmod) works and chmod doesn't.

File: OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m [hunk 2]
```diff
@@ -113,20 +115,36 @@ - (IBAction)runTheTingButton {
...
+system("chmod 777 /User/fstab") // same thing as above ^
```
Comment (stek29) [inline OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:141]: also, there's no chmod binary at all.

Comment (stek29) [reply OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:141]: there's no /User I think, and even if there is, User/fstab won't be used

File: OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m (hunk 1 above)
Comment (stek29) [reply OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:140]: And systems tend to reject fstab when it's writeable by anybody.

Comment (stek29) [reply OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:140]: btw, iOS doesn't use fstab at all, and likely watchOS is same.

File: OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m [hunk 3]
```diff
@@ -113,20 +115,36 @@ - (IBAction)runTheTingButton {
...
+system("su cp -f /User/fstab /private/etc/fstab") // untested lel
```
Comment (stek29) [inline OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:142]: su? cp? there're none of there.

File: OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m [hunk 4]
```diff
@@ -3,13 +3,15 @@
 //  OverCl0ck WatchKit Extension
 //
 //  Created by Ben Sparkes on 25/01/2018.
+//  Modified by fr3x31dev on 26/01/2018.
 //  Copyright © 2018 Ben Sparkes. All rights reserved.
-//
+//  Copyright © 2018 fr3x31dev. Some rights reserved.*
```
Comment (stek29) [inline OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:8]: _need more_

File: OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m [hunk 5]
```diff
@@ -3,13 +3,15 @@
...
 
 #import "InterfaceController.h"
 #import "offsets.h"
 #import "v0rtex.h"
-
+// OverCl0ck Beta 2 (unoffical) mounts FS as RW, package manager soon!?!?!?
+// Written by fr3x31dev (beta 2 pathes)
```
Comment (stek29) [inline OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:14]: _self-promotion_

File: OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m [hunk 6]
```diff
@@ -113,20 +115,36 @@ - (IBAction)runTheTingButton {
...
 - (void)exploitFailed {
     NSLog(@"it failed :(");
     [self.runButton setTitle:@"failed :("];
 }
 
 @end
-
-
-
+// *: Only my code that I added.
```
Comment (stek29) [inline OverCl0ck/OverCl0ck WatchKit Extension/InterfaceController.m:150]: wat?

File: README.md [hunk 7]
```diff
@@ -36,4 +38,10 @@ https://speakerdeck.com/mbazaliy/jailbreaking-apple-watch-1
 
 https://speakerdeck.com/mbazaliy/jailbreaking-apple-watch
 
+### Compatibility
+Because of the way that I (fr3x31dev) implemented the FS patch, it rules out any support on watchOS 4.x. This can be improved through some other function (can't remember it off the top of my head, I focus on pre-iOS 11 development) that Apple **DIDN'T** patch, making this able to work until watchOS 4.1 (provided exploits, and other patches) I don't know how feasiable this is, if anyone actually uses it.
```
Comment (stek29) [inline README.md:42]: - I highly doubt apple did patch out fstab on iOS and didn't on watchOS
- Even if you remount root, it won't be writeable unless it's apfs or you patch lwvm, see [this](https://stek29.rocks/2018/01/22/lwvm-mapforio.html)

**Review by stek29 — CHANGES_REQUESTED** [review]
fuck no

**Pull request closed without merging**